Detectors/CTP/macro/TestFetcher.C

```diff
@@ -0,0 +1,46 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#if !defined(__CLING__) || defined(__ROOTCLING__)
+#include <CCDB/BasicCCDBManager.h>
+#include <DataFormatsCTP/Configuration.h>
+#include <DataFormatsCTP/ctpRateFetcher.h>
+#endif
+using namespace o2::ctp;
+
+void TestFetcher(int runNumber = 556767)
+{
+  auto& ccdb = o2::ccdb::BasicCCDBManager::instance();
+  std::pair<int64_t, int64_t> pp = ccdb.getRunDuration(runNumber);
+  long ts = pp.first + 60;
+  std::cout << "Run duration:" << pp.first << " " << pp.second << std::endl;
+  // Opening run
+  ctpRateFetcher fetcher;
+  fetcher.setupRun(runNumber, &ccdb, ts, 0);
+  ccdb.setURL("http://ali-qcdb-gpn.cern.ch:8083/");
```

Comment:
shouldn't you set the URL before passing the ccdb to `fetcher` ?

Reply:
I think no as in setupRun you want to retrieve objects from CCDB first.

Reply:
ok

Reply:
This may invalidate cached CCDB objects.